Dockerfile

```diff
@@ -4,14 +4,13 @@ ENV PYTHONUNBUFFERED 1
 
 WORKDIR /code
 
-COPY requirements.txt /code/
-COPY dev_requirements.txt /code/
+COPY . /code/
```

Comment:
Com o `COPY` antes to longo bloco de `RUN`, qualquer alteração mínima no projeto vai ser um gatilho para refazer a imagem Docker _toda_. Fica praticamente inviável o uso para desenvolvimento, pois você adiciona o cilco `apt update`, `apt upgrade`, `apt install` e `pip install` a cada vez que for rodar os testes. Normalmente, quando o foco é desenvolvimento, o `COPY` com os arquivos do projeto deve ser uma da últimas coisas no `Dockerfile`.

Então fica a pergunta: o foco é desenvolvimento ou produção?

Reply:
Então, eu só queria mesmo era colocar o `collectstatic` no `RUN`, pra ficar como o jarbas faz, tem alguma sugestão pra voltar ao de antes?

Queria fazer algo parecido com: https://github.com/okfn-brasil/serenata-de-amor/blob/master/Dockerfile#L15

Reply:
Os dois, @cuducos. Além de dar suporte ao desenvolvimento local, a ideia é utilizar o Docker na nossa migração para o Absam (https://github.com/DadosAbertosDeFeira/maria-quiteria/issues/134). Qual seria o lado negativo de copiar por último em produção?

Reply:
> tem alguma sugestão pra voltar ao de antes?

O ganho de rodar o `collectstatic` no _build_ não compensa o tempo de espera em desenvolvimento para rodar a suíte de teste cada vez que tivar um alteração em arquivo do projeto, na minha opinião. Ou seja, primeiro instala as coisas com `apt` e `pip` depois copia os arquivos do projeto. 

> Qual seria o lado negativo de copiar por último em produção?

Não tem lado negativo em copiar por último para produção não : ) Vai ficar uma camada a mais, mas não acho que seja significativo.

Reply:
> Ou seja, primeiro instala as coisas com `apt` e `pip` depois copia os arquivos do projeto.

👍

Reply:
Boa @cuducos ! Concordo contigo. Colocar um passo especifico para copiar apenas o requirement é uma boa, pq isso fica no cache e o dev não precisa esperar. Mesmo para produção isso seria um problema, pois afeta a velocidade do deploy, uma vez que o pull teria um diff maior para baixar.